src/main/java/carpet/script/api/Scoreboards.java

```diff
@@ -396,6 +402,132 @@ public static void apply(Expression expression)
             }
             return LazyValue.TRUE;
         });
+
+        expression.addLazyFunction("bossbar", -1, (c, t, lv) ->
+        {
+            BossBarManager bossBarManager = ((CarpetContext)c).s.getMinecraftServer().getBossBarManager();
+            if(lv.size() > 3) throw new InternalExpressionException("'bossbar' accepts max three arguments");
+            if(lv.size() == 0) {
+                Value ret = ListValue.wrap(bossBarManager.getAll().stream().map(CommandBossBar::getId).map(Identifier::toString).map(StringValue::of).collect(Collectors.toList()));
+                return (_c, _t) -> ret;
+            } else if(lv.size() == 1) {
+                String id = lv.get(0).evalValue(c).getString();
+                Identifier identifier;
+                try {
+                    identifier = new Identifier(id);
+                } catch (InvalidIdentifierException invalidIdentifierException) {
+                    return LazyValue.NULL;
+                }
+                if(bossBarManager.get(identifier) != null) return LazyValue.FALSE;
+                Value ret = StringValue.of(bossBarManager.add(identifier,new LiteralText(id)).getId().toString());
+                return (_c, _t) -> ret;
+            } else {
+                String id = lv.get(0).evalValue(c).getString();
+                Identifier identifier;
+                try {
+                    identifier = new Identifier(id);
+                } catch (InvalidIdentifierException invalidIdentifierException) {
+                    return LazyValue.NULL;
+                }
+                String property = lv.get(1).evalValue(c).getString();
+                CommandBossBar bossBar = bossBarManager.get(identifier);
+                Value propertyValue;
+                if(lv.size() == 3) {
+                    propertyValue = lv.get(2).evalValue(c);
+                } else {
+                    propertyValue = null;
+                }
+                if(bossBar == null) return LazyValue.NULL;
```

Comment:
this can be called right after fetching the bossbar, skipping evaluating lv.get(1) and lv.get(2)